mypy/util.py

```diff
@@ -204,3 +207,91 @@ def replace_object_state(new: object, old: object) -> None:
                         setattr(new, attr, getattr(old, attr))
                     elif hasattr(new, attr):
                         delattr(new, attr)
+
+
+# backport commonpath for 3.4
+if sys.platform == 'win32':
+    def commonpath(paths: Sequence[str]) -> str:
```

Comment:
Do we need the general algorithm here? As I understand we essentially only need something like `is_sub_path(p1, p2)`. Not that I think it is important, but maybe it will allow to simplify code?

Reply:
I was originally hoping to eventually use `os.path.commonpath` once we drop support for 3.4, but I think you are right it would be better to do this ourselves. I will change this when I have time later.